core/src/main/java/io/hyperfoil/core/impl/user/PointPoissonConstantUserGenerator.java

```diff
@@ -0,0 +1,54 @@
+ * {@code T} is a random variable that follows an exponential distribution with rate {@code usersPerSec}.<br><br>
+ * The formula to compute the next fire time is:<br><br>
+ * <code>fireTime(t) = fireTime(t -1) + -log(rand) / usersPerSec</code><br><br>
+ * where the T part (exponential distribution) is computed as {@code -log(rand) / usersPerSec}
+ * and uses the <a href="https://en.wikipedia.org/wiki/Inverse_transform_sampling">inverse transform sampling method<a/>
+ * to generate such intervals following an exponential distribution.<br><br>
+ * Specifically, the exponential distribution is defined as:<br><br>
+ * <code>f(x) = usersPerSec * exp(-usersPerSec * x)</code><br><br>
+ * where {@code x} is the random variable and {@code usersPerSec} is the rate of the distribution.<br><br>
+ * In order to generate samples which follow it we need to compute the inverse function of the cumulative distribution function (CDF).<br><br>
+ * The CDF of the exponential distribution is obtained by integrating the probability density function (PDF) from 0 to {@code x} and is defined as:<br><br>
+ * <code>F(x) = 1 - exp(-usersPerSec * x)</code><br><br>
+ * We can transform this function by performing the logarithm of both sides:<br><br>
+ * <code>log(1 - F(x)) = -usersPerSec * x</code><br><br>
+ * Resolving the previous equation for {@code x} we get:<br><br>
+ * <code>x = -log(1 - F(x)) / usersPerSec</code><br><br>
+ * But given that {@code F(x)} is a random variable that follows a uniform distribution between 0 and 1, {@code 1 - F(x)}
+ * is also a random variable that follows a uniform distribution between 0 and 1.<br><br>
+ * Therefore, the formula to generate samples that follow an exponential distribution is:<br><br>
+ * <code>x = -log(rand) / usersPerSec</code><br><br>
+ * which is the inverse function of the CDF of the exponential distribution and {@code x} is a random variable
+ * that follows a uniform distribution between 0 and 1.<br><br>
+ */
+final class PointPoissonConstantUserGenerator extends PreviousUserAwareGenerator {
+
+    private final Random random;
+    private final double usersPerSec;
+
+    PointPoissonConstantUserGenerator(final double usersPerSec) {
+        this.random = new Random();
+        this.usersPerSec = usersPerSec;
+        this.scheduledUser.fireTime = fireTimeMs(random, usersPerSec, 0);
+    }
+
+    @Override
+    protected double nextFireTimeMs(final double elapsedTimeMs) {
+        return fireTimeMs(random, usersPerSec, elapsedTimeMs);
+    }
+
+    static double fireTimeMs(final Random random, final double usersPerSec, final double elapsedTimeMs) {
+        // the Math.max(1e-20, random.nextDouble()) is to prevent the logarithm to be negative infinity, because
+        // the random number can approach to 0 and the logarithm of 0 is negative infinity
+        return Math.ceil(elapsedTimeMs + 1000 * -Math.log(Math.max(1e-20, random.nextDouble())) / usersPerSec);
+    }
```

Comment:
Given that the static method `fireTimeMs` is not called anywhere else, unless you are planning to do so I think we can remove it and move the logic directly inside `nextFireTimeMs`

```suggestion
    PointPoissonConstantUserGenerator(final double usersPerSec) {
        this.random = new Random();
        this.usersPerSec = usersPerSec;
        this.scheduledUser.fireTime = fireTimeMs(0);
    }

    @Override
    protected double nextFireTimeMs(final double elapsedTimeMs) {
        // the Math.max(1e-20, random.nextDouble()) is to prevent the logarithm to be negative infinity, because
        // the random number can approach to 0 and the logarithm of 0 is negative infinity
        return Math.ceil(elapsedTimeMs + 1000 * -Math.log(Math.max(1e-20, random.nextDouble())) / usersPerSec);
    }
```